packages/cli/src/workflows/workflows.services.ts

```diff
@@ -175,12 +141,24 @@ export class WorkflowsService {
 		}
 
 		filter.id = In(sharedWorkflowIds);
-		return Db.collections.Workflow.find({
+
+		if (typeof filter.name === 'string' && filter.name !== '') {
+			filter.name = Like(`%${filter.name}%`);
+		}
+
+		const findManyOptions: FindManyOptions<WorkflowEntity> = {
 			select,
 			relations,
 			where: filter,
-			order: { updatedAt: 'DESC' },
-		});
+			order: { updatedAt: 'ASC' },
+		};
+
+		if (options?.take) {
+			findManyOptions.skip = options.skip;
+			findManyOptions.take = options.take;
+		}
+
+		return Container.get(WorkflowRepository).findAndCount(findManyOptions);
```

Comment:
Most of this should live in the repository, but let's address that later separately to avoid bloating this PR.